src/pipecat/transports/websocket/fastapi.py

```diff
@@ -294,11 +301,14 @@ async def _receive_messages(self):
                     await self.push_frame(frame)
         except Exception as e:
             logger.error(f"{self} exception receiving data: {e.__class__.__name__} ({e})")
-
-        # Trigger `on_client_disconnected` if the client actually disconnects,
-        # that is, we are not the ones disconnecting.
-        if not self._client.is_closing:
-            await self._client.trigger_client_disconnected()
```

Comment:
I guess that reaching this point means the client has closed the socket, which will trigger `on_client_disconnected`. 

Usually, the task is cancelled(stop) inside the user’s code at that point. Like, inside the `on_client_disconnected` callback.

I’m not sure how the race condition you mentioned would specifically impact this case. Like who is invoking `stop` at the same time ?

Have you been able to reproduce the issue locally ? Do you have the steps we should follow ?

Also tagging @aconchillo here for thoughts.

Reply:
I feel the same. Also. we only want to trigger `on_client_disconnected` when the client really disconnects.